public/app/features/explore/spec/helper/assert.ts

```diff
@@ -4,10 +4,12 @@ import { ExploreId } from '../../../../types';
 
 import { withinExplore } from './setup';
 
-export const assertQueryHistoryExists = (query: string, exploreId: ExploreId = ExploreId.left) => {
+export const assertQueryHistoryExists = async (query: string, exploreId: ExploreId = ExploreId.left) => {
   const selector = withinExplore(exploreId);
 
-  expect(selector.getByText('1 queries')).toBeInTheDocument();
+  await waitFor(() => {
+    expect(selector.getByText('1 queries')).toBeInTheDocument();
```

Comment:
I think you can use `findBy` here with await to get the same thing.

Reply:
Good call. Updated in 9ed4a26d658d6282f03f2dfe2c9205ac4efef0a0